src/features/configTable/configTable.tsx

```diff
@@ -0,0 +1,296 @@
+import * as R from "ramda";
+import {
+  Box,
+  Button,
+  Checkbox,
+  HStack,
+  FormLabel,
+  Input,
+  VStack,
+  Heading,
+  Spacer,
+  IconButton,
+  Table,
+  Thead,
+  Tbody,
+  Tr,
+  Th,
+  Td,
+  FormErrorMessage,
+  FormControl,
+  FormHelperText,
+  Spinner,
+} from "@chakra-ui/react";
+import { FaPlus, FaTrash } from "react-icons/fa";
+import { ValidationErrors } from "final-form";
+import { z } from "zod";
+import { zod2FieldValidator, zod2FormValidator } from "../../lib/zod2form";
+import { usePostConfigMutation } from "./configTableApi";
+import _ from "lodash";
+import errorHandler from "../errorHandler/errorHandler";
+import { dispatchNotification } from "../notifications/notification";
+import { useAppDispatch } from "../../app/hooks";
+import {
+  NotificationDuration,
+  NotificationPosition,
+} from "../notifications/notificationsTypes";
+
+export type Employee = {
+  name: string;
+  surName: string;
+  employed: boolean;
+};
+
+const nameValidator = z
+  .string()
+  .max(10)
+  .refine((x) => !x.endsWith("Kail"), { message: "Kail is not allowed" });
+
+type RowError = {
+  index: number;
+  message: string;
+};
+
+//This is the "whole set" validator for the form. of course for us is a table but it is in fact a form
+//In our configuration pattern this would be primarily used for primary key validation so i made it generic to accept a list of props
+//to check for combinations of duplicates
+const primaryKeyValidator =
+  (propsToCheck: string[]) => (values: { table?: Employee[] }) => {
+    const errors: { table?: { _rowError?: string[] }[] } = {};
+
+    const table = values.table;
+
+    if (!table) {
+      return errors;
+    }
+
+    const propValues = table.map((e) =>
+      propsToCheck.map((prop) => e[prop as keyof Employee]).join("|")
+    );
+    const duplicates: string[] = [];
+
+    propValues.forEach((value, i) => {
+      const duplicateIndexes = propValues.reduce(
+        (indexes, propValue, index) => {
+          if (propValue === value) {
+            indexes.push(index);
+          }
+          return indexes;
+        },
+        [] as number[]
+      );
+
+      if (duplicateIndexes.length > 1 && !duplicates.includes(value)) {
+        duplicates.push(value);
+
+        duplicateIndexes.forEach((index) => {
+          errors.table = errors.table || [];
+          errors.table[index] = errors.table[index] || { _rowError: [] };
+
+          errors.table[index]._rowError!.push(
+            `Duplicate ${propsToCheck.join(
+              ", "
+            )} values are not allowed at indexes: ${duplicateIndexes.join(
+              ", "
+            )}`
+          );
+        });
+      }
+    });
+
+    return errors;
+  };
+
+export default function EditableTableExample(props: {
+  title: string;
+  data: Employee[] | undefined;
+  isLoading: boolean;
+}) {
+  const [
+    postConfig,
+    { isLoading: postConfigIsLoading, isSuccess: postConfigSuccess },
```

Comment:
test a Failure case, in case the Submit fails. you can use 2 separate buttons to call postConfig with 2 different args


await postConfig(values.table, true).unwrap()
or
await postConfig(values.table, false).unwrap()